crates/sui-bridge/src/types.rs

```diff
@@ -1,27 +1,349 @@
+                bytes.push(TOKEN_TRANSFER_MESSAGE_VERSION);
+                // Add nonce
+                bytes.extend_from_slice(&e.nonce.to_le_bytes());
+                // Add source chain id
+                bytes.push(BridgeChainId::SuiTestnet as u8);
+                // Add source tx id length
+                bytes.push(SUI_TX_DIGEST_LENGTH as u8);
+                // Add source tx id
+                bytes.extend_from_slice(a.sui_tx_digest.as_ref());
+                // Add source tx event index
+                bytes.push(a.sui_tx_event_index as u8);
+
+                // Add source address length
+                bytes.push(SUI_ADDRESS_LENGTH as u8);
+                // Add source address
+                bytes.extend_from_slice(&e.sui_address.to_vec());
+                // Add dest chain id
+                bytes.push(BridgeChainId::EthSepolia as u8);
+                // Add dest address length
+                bytes.push(EthAddress::len_bytes() as u8);
+                // Add dest address
+                bytes.extend_from_slice(e.eth_address.as_bytes());
+
+                // Add token id
+                bytes.push(e.token_id as u8);
+
+                // Add token amount
+                bytes.extend_from_slice(&e.amount.to_le_bytes());
+            }
+            BridgeAction::EthToSuiBridgeAction(_e) =>
+            // TODO add formats for other events
+            {
+                unimplemented!()
+            }
+        }
+        bytes
+    }
 }
 
 pub struct BridgeCommitteeValiditySignInfo {
-    pub signatures: HashMap<BridgeCommitteePublicKey, BridgeCommitteeSignature>,
+    pub signatures: HashMap<BridgeAuthorityPublicKey, BridgeAuthoritySignature>,
+}
+
+pub type SignedBridgeAction = Envelope<BridgeAction, BridgeAuthoritySignInfo>;
+pub type VerifiedSignedBridgeAction = VerifiedEnvelope<BridgeAction, BridgeAuthoritySignInfo>;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct BridgeEventDigest(Digest);
+
+impl BridgeEventDigest {
+    pub const fn new(digest: [u8; 32]) -> Self {
+        Self(Digest::new(digest))
+    }
+}
+
+impl Message for BridgeAction {
+    type DigestType = BridgeEventDigest;
+
+    // this is not encoded in message today
+    const SCOPE: IntentScope = IntentScope::BridgeEventUnused;
+
+    // this is not used today
+    fn digest(&self) -> Self::DigestType {
+        unreachable!("BridgeEventDigest is not used today")
+    }
+
+    fn verify_user_input(&self) -> SuiResult {
+        Ok(())
+    }
+
+    fn verify_epoch(&self, _epoch: EpochId) -> SuiResult {
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::types::TokenId;
+    use ethers::types::Address as EthAddress;
+    use fastcrypto::traits::KeyPair;
+    use prometheus::Registry;
+    use sui_types::{
+        base_types::{SuiAddress, TransactionDigest},
+        crypto::get_key_pair,
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_bridge_message_encoding() -> anyhow::Result<()> {
```

Comment:
Since we the solidity code will depend on the encoded data, let's add a regression test (that `to_bytes` for a given set of parameters returns a fixed byte array). The current test will not catch, for example, a change in SUI_TX_DIGEST_LENGTH.